File: src/protocol/web/packet/Timestamp.ts
/*
 * Filename: Timestamp.ts
 * FullPath: apps/CWSP-reborn/src/protocol/web/packet/Timestamp.ts
 * Change date and time: 17.06.00_10.07.2026
 * Reason for changes: Provide a thin canonical epoch-ms timestamp for the web transport.
 *
 * NOTE: CWSP v2 timestamps are Unix epoch milliseconds. `Date.now()` is the canonical
 * source so packet timestamps stay transport-neutral and timezone-free.
 */

/**
 * Returns the current CWSP packet timestamp as Unix epoch milliseconds.
 */
export function nowCwspTimestamp(): number {
    return Date.now();
}

File: src/protocol/java/packet/Timestamp.java
/*
 * Filename: Timestamp.java
 * FullPath: apps/CWSP-reborn/src/protocol/java/packet/Timestamp.java
 * Change date and time: 16.42.00_10.07.2026
 * Reason for changes: Provide millisecond timestamps for packet builders.
 */
package space.u2re.cwsp.protocol.packet;

/**
 * Timestamp helper. CWSP v2 uses epoch milliseconds (matching JS Date.now()).
 */
public final class Timestamp {

    private Timestamp() {
    }

    /** Current epoch time in milliseconds. */
    public static long now() {
        return System.currentTimeMillis();
    }
}
